server/src/main/java/org/candlepin/sync/Exporter.java

```diff
@@ -492,12 +492,18 @@ private void exportProducts(File baseDir, Consumer consumer) throws IOException
             Product product = pool.getProduct();
             products.put(product.getId(), product);
 
+            for (Product providedProduct : product.getProvidedProducts()) {
```

Comment:
Product model translation needs to be n-tier aware during the conversion to a flat list here and below.

Reply:
Fixed.